internal-gateway/Makefile

```diff
@@ -1,21 +1,20 @@
 include ../config.mk
 
-.PHONY: build push deploy
+.PHONY: envoy-config deploy
 
-TOKEN = $(shell cat /dev/urandom | LC_ALL=C tr -dc 'a-z0-9' | head -c 12)
-
-INTERNAL_GATEWAY_IMAGE := $(DOCKER_PREFIX)/internal-gateway:$(TOKEN)
 INTERNAL_IP := $(shell kubectl get secret global-config --template={{.data.internal_ip}} | base64 --decode)
 
-export NAMESPACE := default
-
-build:
-	$(MAKE) -C ../docker hail-ubuntu
-	python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat ../docker/hail-ubuntu-image-ref)'"}}' Dockerfile Dockerfile.out
-	../docker-build.sh . Dockerfile.out $(INTERNAL_GATEWAY_IMAGE)
+envoy-config:
+	HAIL_DOMAIN=$(DOMAIN) python3 ../ci/ci/envoy.py internal-gateway ${HAIL}/letsencrypt/subdomains.txt ${HAIL}/internal-gateway/cds.yaml ${HAIL}/internal-gateway/rds.yaml
+	kubectl -n default apply -f envoy.yaml
```

Comment:
Can we get a NAMESPACE check here like we do in `batch/ deploy`? We could theoretically allow other namespaces for example testing?, but I want to make it explicit from the user and not hard code default here.

Reply:
> We could theoretically allow other namespaces for example testing?

Deploying either of these in namespaces other than default would require allocating a GCP / Azure load balancer and a reserved IP address, which we don't have anything in the Makefiles to support doing that. For this particular target, deploying this into a different namespace would do nothing which I feel is confusing behavior. And we definitely don't want to accidentally point the external load balancer to a different namespace.

If what you're worried about is deploying by accident, would an "Are You Sure?" help? I don't love specifying NAMESPACE when doing anything but default wouldn't make sense, but another option would be to fail if you provide any value other than default

Reply:
Yes an "Are you Sure" would be great! I think you had this functionality somewhere else already. If it's too much work, then don't bother.